alma/lib/Utils/ShareOfCheckoutHelper.php

```diff
@@ -0,0 +1,365 @@
+            ++$ordersByCheckout[$paymentMethod]['orders'][$isoCodeCurrency][self::COUNT_KEY];
+        }
+        foreach ($ordersByCheckout as $paymentKey => $paymentMethodOrders) {
+            $ordersByCheckout[$paymentKey]['orders'] = array_values($paymentMethodOrders['orders']);
+        }
+
+        return array_values($ordersByCheckout);
+    }
+
+    /**
+     * Array structure to send total orders
+     *
+     * @param array $currency
+     *
+     * @return array
+     */
+    private function initTotalOrderResult($currency)
+    {
+        return [
+            self::TOTAL_AMOUNT_KEY => 0,
+            self::TOTAL_COUNT_KEY => 0,
+            self::CURRENCY_KEY => $currency,
+        ];
+    }
+
+    /**
+     * Array structure to send payment method orders
+     *
+     * @param array $currency
+     *
+     * @return array
+     */
+    private function initOrderResult($currency)
+    {
+        return [
+            self::AMOUNT_KEY => 0,
+            self::COUNT_KEY => 0,
+            self::CURRENCY_KEY => $currency,
+        ];
+    }
+
+    /**
+     * Order Ids validated by date
+     *
+     * @return array
+     */
+    public function getOrderIds()
+    {
+        $orderIds = [];
+        $defaultStatesExcluded = [6, 7, 8];
+        $orderIdsByDate = self::getOrdersIdByDate($this->startTime, $this->endTime);
+        foreach ($orderIdsByDate as $orderId) {
+            $currentOrder = new Order($orderId);
+            if (!in_array((int) $currentOrder->current_state, $defaultStatesExcluded)) {
+                $orderIds[] = $currentOrder->id;
+            }
+        }
+
+        return $orderIds;
+    }
+
+    /**
+     * Order by date
+     *
+     * @param string $date_from
+     * @param string $date_to
+     *
+     * @return array
+     */
+    public static function getOrdersIdByDate($date_from, $date_to)
+    {
+        $sql = 'SELECT `id_order`
+                FROM `' . _DB_PREFIX_ . 'orders`
+                WHERE date_add >= \'' . pSQL($date_from) . '\'
+                AND date_add <= \'' . pSQL($date_to) . '\'
+                    ' . Shop::addSqlRestriction();
+        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql);
+
+        $orders = [];
+        foreach ($result as $order) {
+            $orders[] = (int) $order['id_order'];
+        }
+
+        return $orders;
+    }
+
+    /**
+     * @param $startTime
+     *
+     * @return void
+     */
+    public function setShareOfCheckoutFromDate($startTime)
+    {
+        $this->startTime = $startTime . ' 00:00:00';
+        $this->setShareOfCheckoutToDate($startTime);
+    }
+
+    /**
+     * @param $endTime
+     *
+     * @return void
+     */
+    public function setShareOfCheckoutToDate($endTime)
+    {
+        $this->endTime = $endTime . ' 23:59:59';
+    }
+
+    /**
+     * @return string
+     */
+    private function getShareOfCheckoutFromDate()
```

Comment:
```suggestion
    private function getStartDateTime()
```

Here, we are returning a dateTime as string